runtime/stdlib/random.go

```diff
@@ -34,37 +35,164 @@ NOTE: The use of this function is unsafe if not used correctly.
 Follow best practices to prevent security issues when using this function
 `
 
-var revertibleRandomFunctionType = sema.NewSimpleFunctionType(
-	sema.FunctionPurityImpure,
-	nil,
-	sema.UInt64TypeAnnotation,
-)
+var revertibleRandomFunctionType = func() *sema.FunctionType {
+	typeParameter := &sema.TypeParameter{
+		Name:      "T",
+		TypeBound: sema.FixedSizeUnsignedIntegerType,
+	}
+
+	typeAnnotation := sema.NewTypeAnnotation(
+		&sema.GenericType{
+			TypeParameter: typeParameter,
+		},
+	)
+
+	return &sema.FunctionType{
+		TypeParameters: []*sema.TypeParameter{
+			typeParameter,
+		},
+		Parameters: []sema.Parameter{
+			{
+				Identifier:     "modulo",
+				TypeAnnotation: typeAnnotation,
+			},
+		},
+		ReturnTypeAnnotation: typeAnnotation,
+		// `modulo` parameter is optional
+		Arity: &sema.Arity{Min: 0, Max: 1},
+	}
+}()
 
 type RandomGenerator interface {
 	// ReadRandom reads pseudo-random bytes into the input slice, using distributed randomness.
+	// The number of bytes read is equal to the length of input slice.
 	ReadRandom([]byte) error
 }
 
+func getRandomBytes(generator RandomGenerator, numBytes int) []byte {
+	buffer := make([]byte, numBytes)
+
+	var err error
+	errors.WrapPanic(func() {
+		err = generator.ReadRandom(buffer[:])
+	})
+	if err != nil {
+		panic(interpreter.WrappedExternalError(err))
+	}
+
+	return buffer
+}
+
 func NewRevertibleRandomFunction(generator RandomGenerator) StandardLibraryValue {
 	return NewStandardLibraryFunction(
 		"revertibleRandom",
 		revertibleRandomFunctionType,
 		revertibleRandomFunctionDocString,
 		func(invocation interpreter.Invocation) interpreter.Value {
-			return interpreter.NewUInt64Value(
-				invocation.Interpreter,
-				func() uint64 {
-					var buffer [8]byte
-					var err error
-					errors.WrapPanic(func() {
-						err = generator.ReadRandom(buffer[:])
-					})
-					if err != nil {
-						panic(interpreter.WrappedExternalError(err))
-					}
-					return binary.LittleEndian.Uint64(buffer[:])
-				},
-			)
+			inter := invocation.Interpreter
+
+			// TODO: Check if invocation has an argument and implement modulo operation.
+
+			returnIntegerType := invocation.TypeParameterTypes.Oldest().Value
+			returnIntegerStaticType := interpreter.ConvertSemaToStaticType(inter, returnIntegerType)
+
+			switch returnIntegerStaticType {
```

Comment:
Avoid the unnecessary conversion here and simply switch on the sema type instead

Reply:
Done. https://github.com/onflow/cadence/pull/2943/commits/1def84b8a5b61e20a54be4094264da9792bd26e8